stages/org.osbuild.pacman.conf

```diff
@@ -15,24 +15,68 @@ SCHEMA = """
     "type": "string",
     "description": "pacman architecture",
     "default": "x86_64"
+  },
+  "mirrors": {
+    "type": "array",
+    "items": {
+      "type": "string"
+    }
+  },
+  "repositories": {
+    "type": "array",
+    "items": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["name"],
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "siglevel": {
+          "type": "string"
+        },
+        "server": {
+          "type": "string"
+        }
+      }
+    }
   }
 }
 """
 
 
 def main(tree, options):
     arch = options.get("architecture", "x86_64")
+    repositories = options.get("repositories", [])
+    mirrors = options.get("mirrors", [])
     cfg = f"""
 [options]
 Architecture = {arch}
 SigLevel    = Required DatabaseOptional
 LocalFileSigLevel = Optional
 """
+    for repo in repositories:
+        entry = f"[{repo['name']}]\n"
+        if 'siglevel' in repo:
+            entry += f"SigLevel {repo['siglevel']}\n"
+        if 'server' in repo:
+            entry += f"Server = {repo['server']}\n"
+        else:
+            # Fallback for normal repositories
+            entry += "Include = /etc/pacman.d/mirrorlist\n"
+        cfg += entry + "\n"
+
     os.makedirs(os.path.join(tree, "etc"), exist_ok=True)
     cfgpath = os.path.join(tree, "etc", "pacman.conf")
     with open(cfgpath, "w", encoding="utf-8") as cfgfile:
         cfgfile.write(cfg)
 
+    if mirrors:
+        os.makedirs(os.path.join(tree, "etc", "pacman.d"), exist_ok=True)
+        mirrorpath = os.path.join(tree, "etc", "pacman.d", "mirrorlist")
+        with open(mirrorpath, "w", encoding="utf-8") as cfgfile:
+            cfgfile.write("\n".join(f"Server = {m}" for m in mirrors))
```

Comment:
Should this go into a separate stage `org.osbuild.pacman.mirrorlist.conf`?

Reply:
I think it's fine to keep it in the same file.  Maybe @gicmo has a stronger opinion though.

Reply:
if you feel like splitting it, go ahead :)
It does make sense IMO